libs/shared/src/main/java/io/crate/exceptions/Exceptions.java

```diff
@@ -42,6 +43,16 @@ public static void rethrowUnchecked(final Throwable ex) {
     }
 
     public static Exception toException(Throwable t) {
+        if (t instanceof CompletionException || t instanceof ExecutionException) {
```

Comment:
there is also `CancellationException` related to futures' flow - but not sure we need to cover it

Reply:
Afaik it usually isn't raised with a cause, because it is mostly triggered by a `.cancel()` call and not some other exception. It is also a RuntimeException, so it won't get wrapped either.